src/neo4j/api.py

```diff
@@ -114,11 +119,8 @@ def __eq__(self, other):
 # For backwards compatibility
 AuthToken = Auth
 
-# if t.TYPE_CHECKING:
-# commented out as work around for
-# https://github.com/sphinx-doc/sphinx/pull/10880
-# make sure TAuth is resolved in the docs, else they're pretty useless
-_TAuth = t.Union[t.Tuple[t.Any, t.Any], Auth, None]
+if t.TYPE_CHECKING:
+    _TAuth = t.Union[t.Tuple[t.Any, t.Any], Auth, None]
```

Comment:
why for the line is it using union when elsewhere you have used
`typing.Tuple[typing.Any, typing.Any] | Auth | None`

Reply:
Note, that `|` is only used in doc strings. That is because this typing syntax was introduced Python 3.10 ([PEP 604](https://peps.python.org/pep-0604/)). Since the driver's Python baseline is 3.7, I can't use it in code. But sphinx (the docs building tool) does still understand it if it's in docs strings. Might also be because we're building the docs with Python 3.11.